src/UI/class-settings-page.php

```diff
@@ -638,12 +638,19 @@ public function validate_options( array $input ): array {
 		$input['track_page_types'] = self::validate_option_array( $input['track_page_types'] );
 
 		// Detect and prevent duplicate tracking.
-		$duplicate_tracking_found = ! empty( array_intersect( $input['track_post_types'], $input['track_page_types'] ) );
-		if ( $duplicate_tracking_found ) {
+		$duplicate_items = array_intersect( $input['track_post_types'], $input['track_page_types'] );
+		if ( ! empty( $duplicate_items ) ) {
 			add_settings_error(
 				Parsely::OPTIONS_KEY,
 				'track_page_types',
-				__( 'An element is selected twice in post and page types to track. Each element can be selected only once.', 'wp-parsely' )
+				sprintf(
+					/* translators: %s: Item(s) being tracked both as posts and pages. */
+					__(
+						'%s cannot be tracked as both posts and pages. Please select every element only once.',
+						'wp-parsely'
+					),
+					implode( ', ', $duplicate_items )
```

Comment:
Can this be further refined to swap the last `, ` for an ` and `?

Right now it would say: "Foo, Bar cannot be tracked..." or "Foo, Bar, Baz" cannot be tracked...". Ideally, we could reframe it as "Foo and Bar cannot be tracked..." and "Foo, Bar and Baz cannot be tracked...".

Note also that some languages may handle this differently (or may not use `,` as comma punctuation at all), so correct i18n support may need a bit more thinking.

Reply:
We would indeed need more tinkering for real/correct i18n support. Ideally, I would add post types in quotes which I think could make the message even more clear:

> "Foo", "Bar" and "Baz" cannot be tracked...

However, from the 3 languages I know, each one handles quotes differently. So for the time being I stepped back from this. For the time being I'll post a commit to swap the `','` with `and` and we can open a discussion about correct i18n support with the others.

Reply:
In terms of i18n support, I think swapping all commas for an "and" is going to be worse. I'll have a think about this.